softwareComponents/torqueComputation/include/torqueComputation/compute.hpp

```diff
@@ -0,0 +1,150 @@
+        JointsCreation
+    };
+
+    /**
+     * Separate configuration for each component - weights and bounds of each part.
+     */
+    struct ComponentConfiguration {
+        std::vector<double> weights;
+        std::vector<std::optional<double>> bounds;
+    };
+
+    /**
+     * How to process module of desired type - check fromJSON method if new strategy is created.
+     */
+    class VisitModuleStrategy
+    {
+        public:
+            VisitModuleStrategy(const ComponentConfiguration& configuration) 
+                : configuration(configuration) {}
+
+            virtual ~VisitModuleStrategy() {}
+
+            /**
+             * Create all required joints in processed module. Creates links between them and add forces.
+             * @param rofiModule
+             * @param roficomMap
+             * @param joints
+             * @param variableManager
+             * @param gravitation
+             */
+            virtual void createJoints(
+                const rofi::configuration::Module& rofiModule, 
+                const std::unordered_map<std::pair<int, int>, int, IntPairHash>& roficomMap,
+                std::unordered_map<int, Joint*>& joints,
+                VariableManager& variableManager,
+                const arma::vec& gravitation
+            ) const = 0;
+
+            const ComponentConfiguration& getConfiguration() const {return configuration;}
+        
+        private:
+            ComponentConfiguration configuration;
+    };
+
+    /**
+     * Strategy for visiting universal module.
+     */
+    class VisitUniversalModuleStrategy : public VisitModuleStrategy
+    {
+    public:
+        VisitUniversalModuleStrategy(const ComponentConfiguration& configuration) 
+                : VisitModuleStrategy(configuration) {}
+
+        void createJoints(
+            const rofi::configuration::Module& rofiModule, 
+            const std::unordered_map<std::pair<int, int>, int, IntPairHash>& roficomMap,
+            std::unordered_map<int, Joint*>& joints,
+            VariableManager& variableManager,
+            const arma::vec& gravitation
+        ) const;
+    };
+
+    /**
+     * Strategy for visiting pad module.
+     */
+    class VisitPadModuleStrategy : public VisitModuleStrategy
+    {
+    public:
+        VisitPadModuleStrategy(const ComponentConfiguration& configuration) 
+                : VisitModuleStrategy(configuration) {}
+
+        void createJoints(
+            const rofi::configuration::Module& rofiModule, 
+            const std::unordered_map<std::pair<int, int>, int, IntPairHash>& roficomMap,
+            std::unordered_map<int, Joint*>& joints,
+            VariableManager&,
+            const arma::vec& gravitation
+        ) const;
+    };
+
+    /**
+     * Config of torque computation.
+     */
+    struct TorqueConfig {
+        arma::vec gravitation;
+        std::unordered_map<rofi::configuration::ModuleType, VisitModuleStrategy*> visitModuleStrategies;
```

Comment:
Změň typ na `std::unordered_map<rofi::configuration::ModuleType, std::unique_ptr< VisitModuleStrategy > >` a nemusíš řešit správu paměti.